sklearn/tests/metadata_routing_common.py

```diff
@@ -60,7 +60,7 @@ def check_recorded_metadata(obj, method, split_params=tuple(), **kwargs):
         if key in split_params and recorded_value is not None:
             assert np.isin(recorded_value, value).all()
         else:
-            assert recorded_value is value
+            assert np.allclose(recorded_value, value)
```

Comment:
I had to do this, because of a very slight mismatch when testing `OneVsOneClassifier.fit() ` (because of floating-point precision?)

Reply:
I had tried this, because when running `test_setting_request_on_sub_estimator_removes_error` I had gotten the following error:

```
            if key in split_params and recorded_value is not None:
                assert np.isin(recorded_value, value).all()
            else:
>               assert recorded_value is value
E               AssertionError
```

Now I believe it's due to feeding only part of the data into the sub-estumator in `_fit_ovo_binary`. So only `assert np.isin(recorded_value, value).all()` should be checked, not `assert recorded_value is value`.

Setting `"preserves_metadata": "subset",` for ovo does this, but regarding your explanations [here](https://github.com/scikit-learn/scikit-learn/pull/27058#discussion_r1331251536), I still don't understand, why  `partial_fit` didn't raise the same issue, only `fit` did.

Reply:
that makes sense since we're indexing the original array, and the indexing, even though it might select all rows, might be returning a new object (albeit a view to the old one), therefore `is` wouldn't work, and therefore we should set it as a subset in the tests. `partial_fit` might be not doing the sub-selection then.

Reply:
From what I understood, `partial_fit` also does also do a sub-selection. Could you look at the code to tell me if I'm wrong?

Reply:
yes it does. I'm not sure if the test passes then. But I wouldn't worry about it.

Reply:
Yes, the test for  `partial_fit` also passes `assert recorded_value is value`, which it shouldn't.

I will immediately stop worrying about this, but not before telling you that this odd test passing when it shouldn't had lead me totally off track while finding the reasons for `fit` not passing the same test.

The problem is, your code is so flexible, that it's both, hard to read and hard to document.